src/controllers/user/signin.ts

```diff
@@ -1,18 +1,24 @@
-import type { Request, Response } from 'express';
+import type { NextFunction, Request, Response } from 'express';
+import { compare } from 'bcrypt';
 import User from '../../database/user';
+import CustomError from '../../helpers';
 
-const signIn = async (req:Request, res:Response): Promise<void> => {
+const signIn = async (req:Request, res:Response, next: NextFunction): Promise<void> => {
   try {
-    const { username, password } = req.body;
-    const user = await User.findOne({ username });
-    if (user?.password === password) {
-      res.status(200).json({
-        user,
-        msg: 'logged',
-      });
+    const { email, password } = req.body;
+    const user = await User.findOne({ email });

```

Comment:
you have second option which is select filter, you want to get pass word in the select filter with 0 or 1 value because we dont want to bring the password from database